tools/headerparser/headerlib/struct_walker.py

```diff
@@ -0,0 +1,263 @@
+#
```

Comment:
Could you please put all the code you've taken from headerlib to a third_party/ dir and add a README indicating where this code comes from, which version is that, whether there're local modifications etc.?
Take a look at https://cs.chromium.org/chromium/src/third_party/mesa/README.chromium?q=README.chromium&dr=C&l=15 for an example.